src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -0,0 +1,435 @@
+        eventType.setShouldBeRegistered(false);
+
+        List<EventType> existingEventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+        for (EventType existingEventType : existingEventTypes) {
+            if (existingEventType.getName().equals(eventType.getName())) {
+                return null;
+            }
+        }
+
+        JsonArray jsonAttributes = jsonEventType.getAsJsonArray(JSON_ATTRIBUTES_ATTRIBUTE);
+        if (jsonAttributes == null) {
+            return null;
+        }
+        for (JsonElement jsonAttribute : jsonAttributes) {
+            String attributeName = jsonAttribute.getAsJsonObject().get(JSON_NAME_ATTRIBUTE).getAsString();
+            createEventTypeAttribute(eventType, attributeName);
+        }
+
+        String timestampAttributeName = jsonEventType.get(JSON_TIMESTAMP_ATTRIBUTE).getAsString();
+        if (timestampAttributeName == null) {
+            return null;
+        }
+        TypeAttribute timestampAttribute = createEventTypeAttribute(eventType, timestampAttributeName);
+        eventType.setTimeStampAttributeId(timestampAttribute.getId());
+
+        return eventType;
+    }
+
+
+    /**
+     * This method creates a new event attribute and attaches it to an existing event type.
+     * @param eventType - the type which should get expanded
+     * @param attributeName - the name of the new attribute
+     * @return type attribute with given event type and name
+     */
+    private TypeAttribute createEventTypeAttribute(EventType eventType, String attributeName) {
+        TypeAttribute attribute = new TypeAttributeImpl();
+        attribute.setName(attributeName);
+        attribute.setTypeId(eventType.getId());
+
+        if (!PersistenceAdapterImpl.getInstance().saveArtifacts(attribute)) {
+            halt(HttpStatusCodes.ERROR, "event type attribute could not be saved: " + attribute);
+        }
+        return attribute;
+    }
+
+    /**
+     * This method returns all event types with queries that need the given event type.
+     * @param eventType the event type blocking event types are searched for
+     * @return a json string of blocking event types, empty if none
+     */
+    private String getBlockingEventTypes(EventType eventType) {
+        List<EventType> eventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+
+        JsonArray blockingEventTypeIds = new JsonArray();
+        for (EventType type : eventTypes) {
+            if (type.getId() == eventType.getId()) {
+                continue;
+            }
+
+            List<EventQuery> eventQueries = PersistenceAdapterImpl.getInstance().getEventQueries(type.getId());
+            for (EventQuery query : eventQueries) {
+                if (query.getQuery().contains(eventType.getName())) {
+                    blockingEventTypeIds.add(type.getId());
+                }
+            }
+        }
+        if (blockingEventTypeIds.size() == 0) {

```

Comment:
`blockingEventTypeIds` is not a List but a JsonArray, so this method is not applicable.